lib/scanner/xml-scanner/xml-scanner.c

```diff
@@ -97,6 +98,27 @@ tag_matches_patterns(const GPtrArray *patterns, const gint tag_length,
   return FALSE;
 }
 
+
+static GString *
+_pop_text_from_stack(XMLScanner *self)
+{
+  return g_queue_pop_head(self->text_stack);
+}
+
+static void
+_push_current_text_to_stack(XMLScanner *self)
+{
+  g_queue_push_tail(self->text_stack, self->text);
+}
+
+static GString *
+_start_new_text_buffer(XMLScanner *self)
+{
+  if (self->text)
+    _push_current_text_to_stack(self);
+  return scratch_buffers_alloc();
```

Comment:
Thanks, we discussed this in IRL.
As `_start_new_text_buffer()` not just creates a new buffer, but pushes previous ones on the stack (i.e. side effect :) ) it should not look like it's trying to hide this fact.
If it's clearly working through side effects, it will not confuse/hide other functionalities (stack-pushing).

_pop() function can remain as it is currently: we would need a return value anyway to determine the success of popping, doing pop internally and returning a bool value not as useful as returning the popped element itself as return value.
So it's good as it is.